org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Replay/ImportReplay.java

```diff
@@ -400,6 +401,62 @@ public void testIsContentImportable() {
 			assertTrue(isContentImportable(textToPaste));
 		}
 
+		public void testNewSegment() throws IOException, ParseException {
+			final String text = "951212 112700.000 NELSON   @C   22  7  0.63 N 21 45 14.91 W 334.9   2.0      0 \n"
+					+ "951212 112800.000 NELSON   @C   22  7  7.00 N 21 45 19.26 W 340.7   2.0      0 \n"
+					+ "951212 112900.000 NELSON   @C[NEW_SEGMENT]   22  7 14.33 N 21 45 23.01 W 346.1   2.0      0 \n"
+					+ "951212 113000.000 NELSON   @C   22  7 21.35 N 21 45 25.55 W  21.4   2.0      0 \n"
+					+ "951212 113100.000 NELSON   @C   22  7 26.42 N 21 45 22.64 W  21.0   2.0      0 ";
+
+			final ImportReplay importReplay = new ImportReplay();
+			final Layers dummyLayers = new Layers();
+			importReplay.setLayers(dummyLayers);
+			for (final String line : text.split("\n")) {
+				importReplay.readLine(line);
+			}
+			final Enumeration<Editable> elemIterator = dummyLayers.elements();
+			while (elemIterator.hasMoreElements()) {
+				final Editable element = elemIterator.nextElement();
+				assertEquals("Track Imported succesfully for new segment test", "NELSON", element.getName());
+				assertTrue("Correct type in the new segment test", element instanceof TrackWrapper);
+				final TrackWrapper trackWrapperNelson = (TrackWrapper) element;
+				final Enumeration<Editable> legsIterator = trackWrapperNelson.elements();
+				while (legsIterator.hasMoreElements()) {
+					final Editable leg = legsIterator.nextElement();
+					final SegmentList legList = (SegmentList) leg;
+					assertEquals("Amount of segment in new segment test", 2, legList.size());
+				}
+			}
+		}
+
+		public void testNewSegment2() throws IOException, ParseException {
+			final String text = "951212 112700.000 NELSON   @C   22  7  0.63 N 21 45 14.91 W 334.9   2.0      0 \n"
+					+ "951212 112800.000 NELSON   @C[NEW_SEGMENT]   22  7  7.00 N 21 45 19.26 W 340.7   2.0      0 \n"
+					+ "951212 112900.000 NELSON   @C[NEW_SEGMENT]   22  7 14.33 N 21 45 23.01 W 346.1   2.0      0 \n"
+					+ "951212 113000.000 NELSON   @C[NEW_SEGMENT]   22  7 21.35 N 21 45 25.55 W  21.4   2.0      0 \n"
+					+ "951212 113100.000 NELSON   @C   22  7 26.42 N 21 45 22.64 W  21.0   2.0      0 ";
+
+			final ImportReplay importReplay = new ImportReplay();
+			final Layers dummyLayers = new Layers();
+			importReplay.setLayers(dummyLayers);
+			for (final String line : text.split("\n")) {
+				importReplay.readLine(line);
+			}
+			final Enumeration<Editable> elemIterator = dummyLayers.elements();
+			while (elemIterator.hasMoreElements()) {
+				final Editable element = elemIterator.nextElement();
+				assertEquals("Track Imported succesfully for new segment test", "NELSON", element.getName());
+				assertTrue("Correct type in the new segment test", element instanceof TrackWrapper);
+				final TrackWrapper trackWrapperNelson = (TrackWrapper) element;
+				final Enumeration<Editable> legsIterator = trackWrapperNelson.elements();
```

Comment:
@saulhidalgoaular - you can use `getSegments()` to get the list of segments, and to avoid looping through `.elements()`

Reply:
Fixed at https://github.com/debrief/debrief/pull/4934/commits/bd59904a8b48d4a2a9b1050b09d6a5a5a15061dc